File: 88-merge-sorted-array/merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int ptr=nums1.length-1;
        m--;
        n--;
        while(m>=0 ||n>=0)
        {
            if (m<0)
            {
                nums1[ptr]=nums2[n];
                n--;
            }
            else if(n<0)
            {
                nums1[ptr]=nums1[m];
                m--;
            }
            else if(nums1[m]>nums2[n])
            {
                nums1[ptr]=nums1[m];
                m--;
            }else{
                nums1[ptr]=nums2[n];
                n--;
            }
            ptr--;
        }
    }
}

File: 88-merge-sorted-array/merge-sorted-array.go
func merge(nums1 []int, m int, nums2 []int, n int)  {
    ptr1:=m-1
    ptr2:=n-1
    index:=m+n-1
    for index>=0{
        if (ptr1>=0 && ptr2>=0){
            if(nums1[ptr1]>nums2[ptr2]){
                nums1[index]=nums1[ptr1]
                ptr1--
            }else{
                nums1[index]=nums2[ptr2]
                ptr2--
            }
        }else if(ptr1<0){
            nums1[index]=nums2[ptr2]
                ptr2--
        }else{
            nums1[index]=nums1[ptr1]
                ptr1--
        }
        index--
    } 
}